web/.storybook/main.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import path, { dirname, join, resolve } from 'path';
+import { PluginOption, UserConfig, Plugin } from 'vite';
+import type { StorybookConfig } from '@storybook/react-vite';
+import fs from 'fs';
+
+const enterpriseTeleportExists = fs.existsSync(
+  path.join(__dirname, '/../../e/web')
+);
+
+function getAbsolutePath(value: string): any {
+  return dirname(require.resolve(join(value, 'package.json')));
+}
+
+function createStoriesPaths() {
+  const stories = ['../packages/**/*.story.@(ts|tsx)'];
+
+  // include enterprise stories if available (**/* pattern ignores dot dir names)
+  if (enterpriseTeleportExists) {
+    stories.unshift('../../e/web/**/*.story.@(ts|tsx)');
+  }
+
+  return stories;
+}
+
+const rootDirectory = path.resolve(__dirname, '..', '..');
+const webDirectory = path.resolve(rootDirectory, 'web');
+
+const config: StorybookConfig = {
+  stories: createStoriesPaths(),
+  framework: {
+    name: getAbsolutePath('@storybook/react-vite'),
+    options: {
+      builder: {
+        viteConfigPath: resolve(
+          __dirname,
+          '../../web/packages/teleport/vite.config.mts'
```

Comment:
I don't think using Web UI vite config is a good idea. It has a lot of stuff specific to Web UI that we have to disable via env vars or filtering out plugins. Actually, the only thing that we actually need from the Web UI config is `tsconfigPathPlugin` . 
Please take a look at the following simplified config:

```
Subject: [PATCH] Simplify storybook config
---
Index: package.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/package.json b/package.json
--- a/package.json	(revision 6074d7df3b44dc64d5cbb39913d064385c2fc7ae)
+++ b/package.json	(date 1721047194393)
@@ -11,8 +11,8 @@
     "build-term": "yarn workspace @gravitational/teleterm build",
     "start-term": "yarn workspace @gravitational/teleterm start",
     "package-term": "yarn workspace @gravitational/teleterm package",
-    "storybook": "VITE_NO_PROXY=1 storybook dev -p 9002 -c web/.storybook",
-    "storybook-smoke-test": "VITE_NO_PROXY=1 yarn storybook --ci --smoke-test",
+    "storybook": "storybook dev -p 9002 -c web/.storybook",
+    "storybook-smoke-test": "yarn storybook --ci --smoke-test",
     "test": "jest",
     "test-coverage": "jest --coverage && web/scripts/print-coverage-link.sh",
     "test-update-snapshot": "yarn test --updateSnapshot",
@@ -105,5 +105,6 @@
     "workerDirectory": [
       "web/.storybook/public"
     ]
-  }
-}
\ No newline at end of file
+  },
+  "packageManager": "yarn@1.22.22+sha512.a6b2f7906b721bba3d67d4aff083df04dad64c399707841b7acf00f6b133b7ac24255f2652fa22ae3534329dc6180534e98d17432037ff6fd140556e2bb3137e"
+}
Index: web/packages/build/.babelrc.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/web/packages/build/.babelrc.js b/web/packages/build/.babelrc.js
--- a/web/packages/build/.babelrc.js	(revision 6074d7df3b44dc64d5cbb39913d064385c2fc7ae)
+++ b/web/packages/build/.babelrc.js	(date 1721047036622)
@@ -41,6 +41,7 @@
       },
     ],
     ['@babel/preset-react', { runtime: 'automatic' }],
+    '@babel/preset-typescript',
   ];
 }
 
Index: web/.storybook/vite.config.mts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/web/.storybook/vite.config.mts b/web/.storybook/vite.config.mts
new file mode 100644
--- /dev/null	(date 1721047179104)
+++ b/web/.storybook/vite.config.mts	(date 1721047179104)
@@ -0,0 +1,9 @@
+import { defineConfig } from 'vite';
+import { tsconfigPathsPlugin } from '@gravitational/build/vite/tsconfigPaths.mjs';
+
+export default defineConfig(() => ({
+  // We don't use `reactPlugin` (which imports @vitejs/plugin-react-swc under the hood)
+  // because Storybook doesn't support SWC compilter in Vite-based projects.
+  // Once it's available, migrate from babel.
+  plugins: [tsconfigPathsPlugin()],
+}));
Index: web/.storybook/main.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/web/.storybook/main.ts b/web/.storybook/main.ts
--- a/web/.storybook/main.ts	(revision 6074d7df3b44dc64d5cbb39913d064385c2fc7ae)
+++ b/web/.storybook/main.ts	(date 1721047160779)
@@ -16,19 +16,15 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import path, { dirname, join, resolve } from 'path';
-import { PluginOption, UserConfig, Plugin } from 'vite';
+import fs from 'node:fs';
+import path from 'node:path';
+
 import type { StorybookConfig } from '@storybook/react-vite';
-import fs from 'fs';
 
 const enterpriseTeleportExists = fs.existsSync(
   path.join(__dirname, '/../../e/web')
 );
 
-function getAbsolutePath(value: string): any {
-  return dirname(require.resolve(join(value, 'package.json')));
-}
-
 function createStoriesPaths() {
   const stories = ['../packages/**/*.story.@(ts|tsx)'];
 
@@ -40,38 +36,13 @@
   return stories;
 }
 
-const rootDirectory = path.resolve(__dirname, '..', '..');
-const webDirectory = path.resolve(rootDirectory, 'web');
-
 const config: StorybookConfig = {
   stories: createStoriesPaths(),
   framework: {
-    name: getAbsolutePath('@storybook/react-vite'),
-    options: {
-      builder: {
-        viteConfigPath: resolve(
-          __dirname,
-          '../../web/packages/teleport/vite.config.mts'
-        ),
-      },
-    },
+    name: '@storybook/react-vite',
+    options: { builder: { viteConfigPath: 'web/.storybook/vite.config.mts' } },
   },
   staticDirs: ['public'],
-  async viteFinal(config: UserConfig) {
-    config.plugins = config.plugins?.filter((plugin: PluginOption) => {
-      if (plugin && 'name' in plugin) {
-        return (
-          plugin.name !== 'teleport-html-plugin' &&
-          plugin.name !== 'teleport-transform-html-plugin' &&
-          // TODO (avatus) migrate fully to the new react-docgen
-          // see migration guide from 6 -> 8 https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#react-docgen-component-analysis-by-default
-          plugin.name !== 'storybook:react-docgen-plugin'
-        );
-      }
-    });
-
-    return config;
-  },
 };
 
 export default config;
```

Reply:
@gzdunek This patch was missing an empty line before the three backticks, I edited it to include that. Otherwise `git apply` fails with `error: corrupt patch at line 138`.